File: fhs_framework_base_business/src/main/java/com/fhs/ucenter/bean/UcenterFrontUser.java
package com.fhs.ucenter.bean;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fhs.core.base.bean.BaseDO;
import com.fhs.core.group.Add;
import com.fhs.core.group.Update;
import lombok.Builder;
import lombok.Data;
import org.hibernate.validator.constraints.Length;

/**
 * 前端用户表(UcenterFrontUser)实体类
 *
 * @author jackwong
 * @since 2019-03-25 13:58:21
 */

@Data
@Builder
@TableName("t_ucenter_front_user")
public class UcenterFrontUser extends BaseDO<UcenterFrontUser> {
    private static final long serialVersionUID = 545604903343287075L;
    @TableId(value = "user_id", type = IdType.UUID)
    private String userId;
    
    @Length(message = "${column.comment}字段的长度最大为32", groups = {Add.class, Update.class}, max = 32)
    @TableField("user_name")
    private String userName;
    
    //昵称
    @Length(message = "昵称字段的长度最大为32", groups = {Add.class, Update.class}, max = 32)
    @TableField("nick_name")
    private String nickName;
    
    //生日
    @Length(message = "生日字段的长度最大为20", groups = {Add.class, Update.class}, max = 20)
    @TableField("birthday")
    private String birthday;
    
    //手机号
    @Length(message = "手机号字段的长度最大为20", groups = {Add.class, Update.class}, max = 20)
    @TableField("mobile")
    private String mobile;
    
    //性别 1-男 0-女 2-未填写
    @Length(message = "性别 1-男 0-女 2-未填写字段的长度最大为11", groups = {Add.class, Update.class}, max = 11)
    @TableField("sex")
    private String sex;
    
    //用户密码
    @Length(message = "用户密码字段的长度最大为64", groups = {Add.class, Update.class}, max = 64)
    @TableField("passwd")
    private String passwd;
    
    //身份证号
    @Length(message = "身份证号字段的长度最大为32", groups = {Add.class, Update.class}, max = 32)
    @TableField("user_card")
    private String userCard;
    
    //用户来源 0 PC 1微信
    @Length(message = "用户来源 0 PC 1微信字段的长度最大为11", groups = {Add.class, Update.class}, max = 11)
    @TableField("user_resource")
    private String userResource;
    
    //语种
    @Length(message = "语种字段的长度最大为20", groups = {Add.class, Update.class}, max = 20)
    @TableField("language")
    private String language;
    
    //省
    @Length(message = "省字段的长度最大为64", groups = {Add.class, Update.class}, max = 64)
    @TableField("province_id")
    private String provinceId;
    
    //市
    @Length(message = "市字段的长度最大为64", groups = {Add.class, Update.class}, max = 64)
    @TableField("city_id")
    private String cityId;
    
    //区
    @Length(message = "区字段的长度最大为64", groups = {Add.class, Update.class}, max = 64)
    @TableField("area_id")
    private String areaId;
    
    //地址
    @Length(message = "地址字段的长度最大为200", groups = {Add.class, Update.class}, max = 200)
    @TableField("address")
    private String address;
    
    //头像
    @Length(message = "头像字段的长度最大为200", groups = {Add.class, Update.class}, max = 200)
    @TableField("image_path")
    private String imagePath;
    


    //启用标识 0-启用 1-禁用
    @Length(message = "启用标识 0-启用 1-禁用字段的长度最大为1", groups = {Add.class, Update.class}, max = 1)
    @TableField("is_disable")
    private String isDisable;


    

    public UcenterFrontUser(){}
    
     public UcenterFrontUser(
             String userId,
            String userName,
            String nickName,
            String birthday,
            String mobile,
            String sex,
            String passwd,
            String userCard,
            String userResource,
            String language,
            String provinceId,
            String cityId,
            String areaId,
            String address,
            String imagePath,
            String isDisable
     ){
              this.userId=userId;
              this.userName=userName;
              this.nickName=nickName;
              this.birthday=birthday;
              this.mobile=mobile;
              this.sex=sex;
              this.passwd=passwd;
              this.userCard=userCard;
              this.userResource=userResource;
              this.language=language;
              this.provinceId=provinceId;
              this.cityId=cityId;
              this.areaId=areaId;
              this.address=address;
              this.imagePath=imagePath;
              this.isDisable=isDisable;
          }


}

File: fhs_framework_base_business_admin/src/main/webapp/pagex/ucenter/ucenter_front_user.js
var modelConfig= {title:'前端用户',pkey:'user_id',type:'uuid',
    orderBy:'update_time Desc',
    namespace:"ucenter_front_user",table:'t_ucenter_front_user',trans:true};
var listPage={
    listFieldSett:function(){
	  return [
	    {name:'user_name',title:'用户名',width:'10%',align:'center'},
        {name:'nick_name',title:'昵称',width:'10%',align:'center'},
        {name:'birthday',title:'生日',width:'10%',align:'center'},
        {name:'mobile',title:'手机号',width:'15%',align:'center'},
        {name:'sex',title:'性别',width:'10%',align:'center',trans:'book',key:'sex',showField:'transMap.sexName'},
        {name:'address',title:'地址',width:'25%',align:'center'},
        {name:'is_disable',title:'是否启用',width:'9%',formart:'formatRowColor',align:'center',trans:'book',key:'is_disable',showField:'transMap.is_disableName'},
        {name:'create_time',title:'创建时间',width:'10%',align:'center'},
    ]},
  isColumnButton:function(){
	  return  true;
  },
  filters:function(){
      return [
        {name:'nick_name',title:'昵称',type:'input',filterType:'like'},
        {name:'mobile',title:'手机号',type:'input',filterType:'like'},
        {name:'sex',title:'性别',type:'book',code:'sex'},
        {name:'is_disable',title:'是否启用',type:'book',code:'is_disable'},
	 ];
  },
  buttons:function(){
      return [
                  {title:'车辆绑定信息查看',fun:'userBindPlateView',permissionsCode:'ucenter_front_user_bind_plate:see',isRow:true},
                  {title:'启用',fun:'userEnable',permissionsCode:'ucenter_front_user:update',isRow:true},
                  {title:'禁用',fun:'userDisable',permissionsCode:'ucenter_front_user:update',isRow:true}
              ];
  },
  disableButtons:function(){
	    return ["add","update"];
  },
  otherFunctions:function(){
      return {
        userBindPlateView:function(row){
            var _openFrameMsg = {url:'http://park.sxpartner.com//ms/pagex/park/ucenter_front_user_bind_plate_list.jsp?front_user_id=' + row.userId,title:row.nickName + '绑定的车牌'}
            top.postMessage(_openFrameMsg, '*');
        },
        userEnable:function(row){
            if(row.isDisable == "0"){
                EalertE("已经在启用中");
                return;
            }
            $.ajax({
    　　　　　　url:'http://base.sxpartner.com//ms/x/ucenter_front_user/update/'+row.userId,
    　　　　　　data:{"isDisable":"0"},
    　　　　　　type:'POST',
    　　　　　　success:function(str){	//成功回调函数
                    Ealert("状态修改成功");
                    reload();
    　　　　　　},
    　　　　　　error:function (err){	//失败回调函数
                    EalertE("状态修改失败");
    　　　　　　}
    　　　　});
        },
        userDisable:function(row){
            if(row.isDisable == "1"){
                EalertE("已经在禁用中");
                return;
            }
            $.ajax({
    　　　　　　url:'http://base.sxpartner.com//ms/x/ucenter_front_user/update/'+row.userId,
    　　　　　　data:{"isDisable":"1"},
    　　　　　　type:'POST',
    　　　　　　success:function(str){	//成功回调函数
                    Ealert("状态修改成功");
                    reload();
    　　　　　　},
    　　　　　　error:function (err){	//失败回调函数
                    EalertE("状态修改失败");
    　　　　　　}
    　　　　});
        }
	  }
  }
};
var add={
	formFields:function(){
	     return [
            {name:'user_name',title:'用户名',type:'input'},
            {name:'nick_name',title:'昵称',type:'input'},
            {name:'birthday',title:'出生日期',type:'date'},
            {name:'mobile',title:'手机号',type:'input',dataType:'m'},
            {name:'sex',title:'性别',type:'book',code:'sex'},
            {name:'passwd',title:'密码',type:'password'},
            {name:'user_card',title:'身份证号',type:'input'},
            {name:'user_resource',title:'用户来源',type:'book',code:'user_resource'},
            {name:'language',title:'语种',type:'input'},
            {name:'image_path',type:'up',title:'头像',placeholder:'请上传头像图片'},
            {name:'is_disable',title:'是否启用',type:'book',code:'is_disable'},
            {name:'province_id',title:'省',type:'input'},
            {name:'city_id',title:'市',type:'input'},
            {name:'area_id',title:'区',type:'input'},
            {name:'address',title:'地址',type:'input'},
		 ];
	},
    otherFunctions:function(){
      return {
	     ready:function(){
	    },
	    loadSuccess:function(info){

	    },
	    onSave:function(){

	    },
		saveSucess:function(){
	    },
		saveError:function(){

	    },
	  }
   }
}